### ISSUE TITLE
Import macro from string

### ISSUE THREAD

**Issue by yalg** [twigphp/Twig#2946]
Hi

I've got a problem since v2.7.0 with template_from_string and macro.

Here is my test sample to understand :

```
{% import template_from_string("{% macro test() %} Imported macro successfull {% endmacro %}") as importedMacros %}
{{ importedMacros.test() }}
```

v2.6.3 : SUCCESS
v2.7.0 : FAIL with error 

> Attempted to call an undefined method named "macro_test" of class "Twig\TemplateWrapper".

Is there a thing that I miss since 2.7 ?

Thanks!

**Comment by stof** [comment]
why would you put the macro in a string template that you would need to compile at runtime ?

**Comment by stof** [comment]
@fabpot should the `import()` tag unwrap a TemplateWrapper to get the actual template ?

**Comment by yalg** [comment]
@stof it is "User defined templates" stored in database used by Html2Pdf.

For some big templates with iterative pages, it's usefull to separate each iterative page in separate template, and put it in a macro. Then we call them by a TwigFunction having the template's unique short name as argument.

If there is another more usefull tool, I whould be happy to know which one! :)

**Comment by fabpot** [comment]
fixed by #2951

**Comment by yalg** [comment]
Thanks @fabpot !
